caproto/ioc_examples/ternaryarray.py

```diff
@@ -0,0 +1,143 @@
+import asyncio
+
+from functools import partial
+from collections import OrderedDict
+from caproto.server import PVGroup, ioc_arg_parser, pvproperty, run
+from ophyd import EpicsSignal, EpicsSignalRO, PVPositionerPC
+from ophyd import Component as Cpt
+
+
+class TernaryDeviceSim:
+    """
+    A device with three states.
+
+    Parameters
+    ----------
+    delay: float, optional
+        The time it takes for the device to change from state-0 to state-1.
+    """
+
+    def __init__(self, delay=0.5):
+        self._delay = delay
+        self._state = None
+
+    async def set(self):
+        if not self._state:
+            self._state = None
+            await asyncio.sleep(self._delay)
+            self._state = True
+
+    async def reset(self):
+        if self._state:
+            self._state = None
+            await asyncio.sleep(self._delay)
+            self._state = False
+
+    @property
+    def state(self):
+        return self._state
+
+
+class TernaryArrayIOC(PVGroup):
+    """
+    Example IOC that has an array of TernaryDevices.
+
+    Parameters
+    ----------
+    count: integer
+        The number of devices in the array.
+    """
+
+    def __init__(self, count=10, *args, **kwargs):
+        self._devices = [TernaryDeviceSim() for i in range(count)]
+
+        # Dynamically setup the pvs.
+        for i in range(count):
```

Comment:
Do this as a helper function that uses `type` to define a new class.  Going up (down?) an abstraction layer will simplify this a a bunch.